enumfields/fields.py

```diff
@@ -179,26 +224,25 @@ def _get_all_parent_choices(self, supvalue):
             choice for choice in self.enum if supvalue in choice.parents
         }
 
-    def clean(self, value, model_instance):
-        if self._get_all_parent_choices(self._get_supvalue(model_instance)):
-            return super().clean(value, model_instance)
-        else:
-            return None
-
-    def _raise_error_if_value_should_be_empty(self, value, supvalue):
+    def _validate_parent_value_empty(self, value, supvalue):
         if supvalue not in self._get_all_parent_values() and value is not None:
             raise ValidationError(ugettext('Value must be empty'))
 
-    def _raise_error_if_value_not_allowed(self, value, supvalue):
+    def _validate_parent_value(self, value, supvalue):
         allowed_values = self._get_all_parent_choices(supvalue)
         if allowed_values and value not in allowed_values:
             raise ValidationError(ugettext('Allowed choices are {}.').format(
                 ', '.join(('{} ({})'.format(*(val.label, val)) for val in allowed_values))
             ))
 
     def validate(self, value, model_instance):
-        self._raise_error_if_value_should_be_empty(value, self._get_supvalue(model_instance))
-        self._raise_error_if_value_not_allowed(value, self._get_supvalue(model_instance))
+        parent_field_value = self._get_supvalue(model_instance)
+        if self._get_all_parent_choices(parent_field_value):
+            self._validate_parent_value_empty(value, parent_field_value)
+            self._validate_parent_value(value, parent_field_value)
+        self._validate_inital_value(value, model_instance)
+        self._validate_next_value(value, model_instance)
+        super(EnumSubFieldMixin, self).validate(value, model_instance)
```

Comment:
you can remove EnumSubFieldMixin, self (python3) super().

Reply:
good. Sometimes I don't know whether I should support also Python 2 so I rather do that ugly version. Changed